src/lib/dcgs.pl

```diff
@@ -216,7 +219,7 @@
 
 user:term_expansion(Term0, Term) :-
     nonvar(Term0),
-    dcg_rule(Term0, Term).
+    catch(dcg_rule(Term0, Term), E, (Term = null, format("   ~q~n", [E]))).
```

Comment:
A blind catchall just leads to further problems. So restrict it to those terms that make sense here. Further, note that many implementations permit lists of clauses including `[]`.

Reply:
Yes, I would also like to use an empty list, but currently Scryer just compiles a clause `[]/0`, which isn't what I want to achieve here. Maybe that is a more deep problem to be solved instead.

Reply:
One can always define the fact `[]` with the rule `[] :- true.`

Reply:
... and, it is still a catchall.

Reply:
Oh... I forgot to push a commit